lib/cadet/jobs/autograder/plagiarism_checker.ex

```diff
@@ -0,0 +1,70 @@
+defmodule Cadet.Autograder.PlagiarismChecker do
+  @moduledoc """
+  This module performs the plagiarism check script and sends results to
+  Amazon S3.
+  """
+  use Cadet, :context
+  use Que.Worker
+
+  require Logger
+
+  import Ecto.Query
+
+  @bucket_name :cadet
+               |> Application.fetch_env!(:plagiarism_check_vars)
+               |> Keyword.get(:bucket_name)
+
+  @plagiarism_script_path :cadet
+                          |> Application.fetch_env!(:plagiarism_check_vars)
+                          |> Keyword.get(:plagiarism_script_path)
+
+  def perform(assessment_id) when is_ecto_id(assessment_id) do
+    Logger.info("Running plagiarism check on Assessment #{assessment_id}")
+
+    {_, exit_result} =
```

Comment:
Refer to https://hexdocs.pm/elixir/System.html#cmd/3. Look at the typespec: actually the return tuple is `{output, exit_code}`

So perhaps what you want is `{result, 0} = System.cmd...`

It's your choice whether to pattern match exit_code with `0` or not. If you do, it's in line with Erlang's let it crash philosophy -- will be captured by sentry. Else you should instead do sth like this:
```elixir
case System.cmd(...) do
  {result, 0} ->
    ...
  _ ->
    # handle error
end
```

Reply:
yeah, this looks sensible. Will it make sense to output something like a json string as the `result`, and then parse into a `Map`? might want to get multiple value results from the python script.

Reply:
So the python script outputs json?

Reply:
not yet, but I can make it do so. Will probably be doing that as there are multiple pieces of info to return.